relay-spans/src/span.rs

```diff
@@ -473,14 +564,81 @@ mod tests {
                     "value": {
                         "stringValue": "my 1st transaction"
                     }
+                },
+                {
+                    "key": "sentry.measurements.memory.value",
+                    "value": {
+                        "doubleValue": 9001.0
+                    }
+                },
+                {
+                    "key": "sentry.measurements.memory.unit",
+                    "value": {
+                        "stringValue": "byte"
+                    }
+                },
```

Comment:
I made `.value` and `.unit` separate attributes to reduce nesting. But I'm open to the alternative: making `sentry.measurements.memory` an attribute with a `kvlistValue` inside (like the metrics summary below).

Reply:
@cleptric @phacops looking for opinions here.